app/routes/helper_functions.py

```diff
@@ -0,0 +1,296 @@
+
+    instance = [instance.to_json() for instance in instances]
+
+    return jsonify(instance), HTTPStatus.OK	 
+
+
+def get_one_instance(cls, id):
+    instance = get_model_by_id(cls, id)
+    return create_response(cls, instance)
+
+
+def update_instance(cls, id):
+    instance = get_model_by_id(cls, id)
+    instance_info = request.get_json()
+
+    if 'title' in instance_info:
+        instance.title = instance_info['title']
+
+    if hasattr(cls, 'description'):
+        if 'description' in instance_info:
+            instance.description = instance_info['description']
+        elif 'completed_at' in instance_info:
+            instance.completed_at = instance_info['completed_at']
+        else:
+            instance.completed_at = None
+
+    return create_response(cls, instance)
+
+
+def delete_instance(cls, id):
+    instance = get_model_by_id(cls, id)
+
+    db.session.delete(instance)
+    db.session.commit()
+
+    message = f'{cls.__name__} {id} "{instance.title}" successfully deleted'
+
+    return make_response({"details": message}, HTTPStatus.OK)
+
+
+def make_instance_complete(cls, id):
+    instance = get_model_by_id(cls, id)
+
+    setattr(instance, "completed_at", NOWTIME)
+
+    slack_api_url = "https://slack.com/api/chat.postMessage"
+
+    slack_request_body = {
+        "channel": CHANNEL,
+        "text": f"Someone just completed the task {instance.title}"
+    }
+    slack_request_headers = {
+        "Authorization": f"Bearer " + SLACK_TOKEN
+    }
+
+    slack_response = requests.post(slack_api_url, json=slack_request_body, headers=slack_request_headers)
+
+    return create_response(cls, instance)
+
+
+def make_instance_incomplete(cls, id):
+    instance = get_model_by_id(cls, id)
+
+    setattr(instance, "completed_at", None)
+
+    return create_response(cls, instance)
+
+
+# Working on making this agnostic to being added to goals so that tasks can be added to project, lists, etc
+def add_tasks_to_model(cls, id):
+    model = get_model_by_id(cls, id)
+    request_body = request.get_json()
+    task_ids = request_body["task_ids"]
+
+    for task in model.tasks:
+        task.goal_id = None
+
+    for task_id in task_ids:
+        task = get_model_by_id(Task, task_id)
+        task.goal_id = model.goal_id
+    
+    goal_id = model.goal_id
+
+    db.session.commit()
+
+    return make_response({"id": goal_id, "task_ids": task_ids}, HTTPStatus.OK)
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+# goal_bp.route("/<goal_id>/tasks", methods=["POST"])
+# def post_task_to_goal(goal_id):
+#     goal = validate_model(Goal, goal_id)
+#     request_body = request.get_json()
+#     for task_id in request_body["task_id"]:
+#         task = validate_model(Task, task_id)
+#         task.goal_id = goal.goal_id
+#     db.session.commit()
+#     task_id_list = []
+#     for task_id in goal.tasks:
+#         task_id_list.append(task_id)
+
+#     response_body = {
+#         "id": goal.goal_id,
+#         "task_ids": [task_id_list]
+#     }
```

Comment:
Don't forget to delete commented code before submitting a PR! It's ok to refer to it while in development, but after, you should rely on commit histories to see what you've done!